### PR TITLE
application: serial_lte_modem: Switch to picolibc

### PR DESCRIPTION
**Pull request by junqingzou**
Switch from newlib_libc to picolibc to save flash usage.

### CODE REVIEW

**Comment by MarkusLassila** [comment]
@junqingzou: Compilation fails with:
arm-zephyr-eabi-gcc: fatal error: cannot read spec file 'picolibc.specs': No such file or directory

**Comment by junqingzou** [comment]
> @junqingzou: Compilation fails with: arm-zephyr-eabi-gcc: fatal error: cannot read spec file 'picolibc.specs': No such file or directory

Thanks, [NCSIDB-1039](https://nordicsemi.atlassian.net/browse/NCSIDB-1039)

**Review by SeppoTakalo — APPROVED** [review]

File: applications/serial_lte_modem/prj.conf [hunk 1]
```diff
@@ -7,8 +7,7 @@
 CONFIG_LOG=y
 CONFIG_LOG_DEFAULT_LEVEL=3
 CONFIG_STACK_SENTINEL=y
-CONFIG_NEWLIB_LIBC=y
-CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+CONFIG_PICOLIBC=y
```
Comment (MarkusLassila) [inline applications/serial_lte_modem/prj.conf:10]: This cannot be merged before we update our test environments.

**Review by MarkusLassila — CHANGES_REQUESTED** [review]
No changes needed in this PR, but cannot be merged at this point.

File: applications/serial_lte_modem/prj.conf (hunk 1 above)
Comment (junqingzou) [reply applications/serial_lte_modem/prj.conf:10]: > This cannot be merged before we update our test environments.

SLM CI does not complete while the job was submitted yesterday.
Previous, completed with environmental problem (Zephyr SDK version)

**Comment by junqingzou** [comment]
Close due to cannot pass CI

**Pull request closed without merging**